packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -626,6 +601,70 @@ export class TreeCheckout implements ITreeCheckoutFork {
 		}
 	}
 
+	private createRevertible(
+		revision: RevisionTag,
+		kind: CommitKind,
+		checkout: TreeCheckout = this,
+		onRevertibleDisposed?: (revertible: Revertible) => void,
+	): ClonableRevertible {
+		const commitBranches = checkout.revertibleCommitBranches;
+
+		const revertible: ClonableRevertible = {
+			get status(): RevertibleStatus {
+				const revertibleCommit = commitBranches.get(revision);
+				return revertibleCommit === undefined
+					? RevertibleStatus.Disposed
+					: RevertibleStatus.Valid;
+			},
+			revert: (release: boolean = true) => {
+				if (revertible.status === RevertibleStatus.Disposed) {
+					throw new UsageError("Unable to revert a revertible that has been disposed.");
+				}
+
+				const revertMetrics = checkout.revertRevertible(revision, kind);
+				checkout.logger?.sendTelemetryEvent({
+					eventName: TreeCheckout.revertTelemetryEventName,
+					...revertMetrics,
+				});
+
+				if (release) {
+					revertible.dispose();
+				}
+			},
+			clone: (forkedBranch?: TreeBranch) => {
+				return checkout.cloneRevertible(revision, kind, forkedBranch);
+			},
+			dispose: () => {
+				if (revertible.status === RevertibleStatus.Disposed) {
+					throw new UsageError(
+						"Unable to dispose a revertible that has already been disposed.",
+					);
+				}
+				checkout.disposeRevertible(revertible, revision);
+				onRevertibleDisposed?.(revertible);
+			},
+		};
+
+		return revertible;
+	}
+
+	private cloneRevertible(
```

Comment:
The logic in this method is simple enough that I would just get rid of it and inline the code in the one place that it is called. If you do that, it will also simplify the logic because you always know that `branch` is defined (it's `forkedBranch`).